src/ArrayHelper.php

```diff
@@ -180,7 +180,10 @@ public static function getValue($array, $key, $default = null)
             $key = $lastKey;
         }
 
-        if (is_array($array) && (isset($array[$key]) || array_key_exists($key, $array))) {
+        if (is_float($key)) {
```

Comment:
Why did you add it?

Reply:
https://github.com/yiisoft/yii2/pull/11549